sdk/eventhub/azure-eventhub/migration_guide.md

```diff
@@ -0,0 +1,250 @@
+
+## Migration samples
+
+* [Receiving events](#migrating-code-from-eventhubclient-and-receiverasync-to-eventhubconsumerclient-for-receiving-events)
+* [Receiving events with checkpointing](#migrating-code-from-eventhubclient-and-senderasync-to-eventhubproducerclient-for-sending-events)
+* [Sending events](#migrating-code-from-eventprocessorhost-to-eventhubconsumerclient-for-receiving-events)
+
+### Migrating code from `EventHubClient` and `ReceiverAsync` to `EventHubConsumerClient` for receiving events
+
+In V1, `ReceiverAsync.receive()` returns a list of EventData.
+
+In V5, EventHubConsumerClient.receive() calls user callback on_event to process events.
+
+For example, this code which keeps receiving from a partition in V1:
+
+```python
+client = EventHubClientAsync.from_connection_string(connection_str)
+receiver = client.add_async_receiver(consumer_group="$default", partition="0", offset=Offset('@latest'))
+try:
+    await client.run_async()
+    logger = logging.getLogger("azure.eventhub")
+    while True:
+        received = await receiver.receive(timeout=5)
+        for event_data in received:
+            logger.info("Message received:{}".format(event_data.body_as_str()))
+finally:
+    await client.stop_async()
+```
+
+Becomes this in V5:
+
+```python
+logger = logging.getLogger("azure.eventhub")
+async def on_event(partition_context, event):
+    logger.info("Message received:{}".format(event.body_as_str()))
+
+client = EventHubConsumerClient.from_connection_string(
+    conn_str=CONNECTION_STR, consumer_group="$default", eventhub_name=EVENTHUB_NAME
+)
+async with client:
+    await client.receive(on_event=on_event, partition_id="0", starting_position="@latest")
+```
+
+### Migrating code from `EventHubClient` and `SenderAsync` to `EventHubProducerClient` for sending events
+
+In V1, you create an `EventHubClient`, then create a `SenderAsync`, and call `SenderAsync.send` to send an event that may have
+a list/generator of messages.
+
+In V5, this has been consolidated into a one method - `EventHubProducerClient.send_batch`.
+Batching merges information from multiple events into a single send, reducing
+the amount of network communication needed vs sending events one at a time.
+This method deterministically tells you whether the batch of events are sent to the event hub.
+
+So in V1:
+```python
+client = EventHubClientAsync.from_connection_string(connection_str)
+sender = client.add_async_sender(partition="0")
+try:
+    await client.run_async()
+    event_data = EventData(b"A single event")
+    await sender.send(event_data)
+finally:
+    await client.stop_async()
+```
+
+In V5:
+```python
+producer = EventHubProducerClient.from_connection_string(conn_str=EVENT_HUB_CONNECTION_STR, eventhub_name=EVENTHUB_NAME)
```

Comment:
same comment for the `eventhub`/`eventhub_name`

Reply:
Change to use eventhub=EVENTHUB_NAME for v1